### ISSUE TITLE
RangeError (end): Invalid value: Not in inclusive range 0..1: 2

### ISSUE THREAD

**Issue by tmaihoff** [hadukin/custom_sliding_segmented_control#22]
Using the last element as initial value throws a `RangeError`.

Example project reproducing the error:
https://github.com/tmaihoff/sccs_issue

In the following snippet, the first `CustomSlidingSegementedControl` works fine, the second throws the error:
```dart
CustomSlidingSegmentedControl<int>(
  initialValue: 2, // works fine
  children: const {
    1: Text('News'),
    2: Text('Map'),
    3: Text('Paper'),
  },
  decoration: BoxDecoration(
    color: CupertinoColors.lightBackgroundGray,
    borderRadius: BorderRadius.circular(8),
  ),
  thumbDecoration: BoxDecoration(
    color: Colors.white,
    borderRadius: BorderRadius.circular(6),
    boxShadow: [
      BoxShadow(
        color: Colors.black.withOpacity(.3),
        blurRadius: 4.0,
        spreadRadius: 1.0,
        offset: const Offset(
          0.0,
          2.0,
        ),
      ),
    ],
  ),
  duration: const Duration(milliseconds: 300),
  curve: Curves.easeInToLinear,
  onValueChanged: (v) {
    print(v);
  },
),
CustomSlidingSegmentedControl<int>(
  initialValue: 2, // throws range error
  children: const {
    0: Text('News'),
    1: Text('Map'),
    2: Text('Paper'),
  },
  decoration: BoxDecoration(
    color: CupertinoColors.lightBackgroundGray,
    borderRadius: BorderRadius.circular(8),
  ),
  thumbDecoration: BoxDecoration(
    color: Colors.white,
    borderRadius: BorderRadius.circular(6),
    boxShadow: [
      BoxShadow(
        color: Colors.black.withOpacity(.3),
        blurRadius: 4.0,
        spreadRadius: 1.0,
        offset: const Offset(
          0.0,
          2.0,
        ),
      ),
    ],
  ),
  duration: const Duration(milliseconds: 300),
  curve: Curves.easeInToLinear,
  onValueChanged: (v) {
    print(v);
  },
),
```

**Comment by hadukin** [comment]
@tmaihoff hi, i will try to fix the problems soon

**Comment by hadukin** [comment]
@tmaihoff fixed in version 1.7.3
